Sources/NIO/ByteBuffer-views.swift

```diff
@@ -16,48 +16,161 @@
+                    let endOfRangeToReplace = subrange.upperBound
+                    let srcPtr = ptr.baseAddress!.advanced(by: endOfRangeToReplace)
+
+                    let endOfReplacedRange = subrange.lowerBound.advanced(by: newElements.count)
+                    let dstPtr = ptr.baseAddress!.advanced(by: endOfReplacedRange)
+
+                    let numberOfBytesToMove = endOfRangeToReplace.distance(to: oldWriterIndex)
+                    memmove(dstPtr, srcPtr, numberOfBytesToMove)
+                }
+            }
+        } else {
+            // We have more elements to write in the subrange than the subrange currently has space
+            // for: widen the capacity of the underlying buffer.
+            let additionalByteCount = newElements.count - subrange.count
+            self._buffer.reserveCapacity(self._buffer.readableBytes + additionalByteCount)
+
+            // We have capacity but we still need to shift everything after the subrange forwards.
+            let writerIndex = self._buffer.writerIndex
+            self._buffer.withVeryUnsafeMutableBytes { ptr in
+                let startOfRangeToMove = subrange.endIndex
+                let srcPtr = ptr.baseAddress!.advanced(by: startOfRangeToMove)
+
+                let startOfMovedRange = startOfRangeToMove.advanced(by: additionalByteCount)
+                let dstPtr = ptr.baseAddress!.advanced(by: startOfMovedRange)
+
+                let numberOfBytesToMove = startOfRangeToMove.distance(to: writerIndex)
+                memmove(dstPtr, srcPtr, numberOfBytesToMove)
+            }
+
+            // We added extra bytes before the writer index, move it forwards accordingly.
+            self._buffer.moveWriterIndex(to: writerIndex + additionalByteCount)
+
+            // Update our own range, which has now grown.
+            self._range = self._range.startIndex ..< self._range.endIndex.advanced(by: additionalByteCount)
+
+            // We've made space for the extra bytes, we've updated the buffer and our own range, all
+            // that's left to do is to replace the bytes in the subrange with the new elements.
+            let rangeToReplace = subrange.startIndex ..< subrange.endIndex.advanced(by: additionalByteCount)
+            self._copyBytes(to: rangeToReplace, from: newElements)
+        }
+    }
+
+    /// Copy bytes from `newElements` to the given `subrange`.
+    ///
+    /// - Precondition: `newElements.count` must be less than or equal to `subrange.count`.
+    /// - Parameter subrange: The subrange of the view into why the elements should be copied.
+    /// - Parameter newElements: The elements to copy into the subrange.
+    @inlinable
+    /* private but inlinable */ mutating func _copyBytes<C: Collection>(to subrange: Range<Index>, from newElements: C) where ByteBufferView.Element == C.Element {
+        precondition(newElements.count <= subrange.count, "subrange must not be longer than newElements")
```

Comment:
isn't this whole function just

    self._buffer.setBytes(newElements, at: subrange.startIndex)

?

Reply:
Ah, yup, it is!